边json数据导入格式不统一时数组溢出

# 1. 问题
每行json边数据的kv对数量不统一时会导致数组溢出，导入中断

（点数据无此问题）

# 2. 报错
```java
2021-05-12 16:19:57 [main] [ERROR] c.b.h.l.u.Printer [] - Failed to load, cause: 6
java.lang.ArrayIndexOutOfBoundsException: 6
        at com.baidu.hugegraph.loader.builder.ElementBuilder$VertexPkKVPairs.extractFromEdge(ElementBuilder.java:522) ~[classes/:?]
        at com.baidu.hugegraph.loader.builder.EdgeBuilder.build(EdgeBuilder.java:76) ~[classes/:?]
        at com.baidu.hugegraph.loader.task.ParseTaskBuilder.lambda$buildTask$0(ParseTaskBuilder.java:98) ~[classes/:?]
        at com.baidu.hugegraph.loader.task.ParseTaskBuilder$ParseTask.get(ParseTaskBuilder.java:157) ~[classes/:?]
        at com.baidu.hugegraph.loader.HugeGraphLoader.executeParseTask(HugeGraphLoader.java:268) ~[classes/:?]
        at com.baidu.hugegraph.loader.HugeGraphLoader.load(HugeGraphLoader.java:247) ~[classes/:?]
        at com.baidu.hugegraph.loader.HugeGraphLoader.load(HugeGraphLoader.java:198) ~[classes/:?]
        at com.baidu.hugegraph.loader.HugeGraphLoader.loadInputs(HugeGraphLoader.java:171) ~[classes/:?]
        at com.baidu.hugegraph.loader.HugeGraphLoader.load(HugeGraphLoader.java:109) [classes/:?]
        at com.baidu.hugegraph.loader.HugeGraphLoader.main(HugeGraphLoader.java:72) [classes/:?]
```

# 3.  问题数据
```JavaScript
//此处有"protocol": ["http"]
{"src": ["sandbox"], "domain": "crl.verisign.com", "protocol": ["http"], "first_seen": "2021-05-02 06:06:15", "last_seen": "2021-05-02 06:06:15", "port": ["80"], "md5": "a22d52dfbf06fcee549003d50413bc71"}    
//此行无"protocol": ["http"]
{"src": ["sandbox"], "domain": "crl3.digicert.com", "first_seen": "2021-05-02 06:06:15", "last_seen": "2021-05-02 06:06:15", "port": ["80"], "md5": "a22d52dfbf06fcee549003d50413bc71"} 
```
将第二行补充上属性数据"protocol": ["http"]，则可导入成功

# 4. 问题代码
`this.vertexIdsIndex.sourceIndexes ` 和`this.vertexIdsIndex.targetIndexes `只计算了一次 ,不同行不应该复用
```java
    public List<Edge> build(String[] names, Object[] values) {
        if (this.vertexIdsIndex == null) {
            this.vertexIdsIndex = this.extractVertexIdsIndex(names);
        }
        EdgeKVPairs kvPairs = this.newEdgeKVPairs();
        kvPairs.source.extractFromEdge(names, values,
                                       this.vertexIdsIndex.sourceIndexes);
        kvPairs.target.extractFromEdge(names, values,
                                       this.vertexIdsIndex.targetIndexes);
```

--- a/src/main/java/com/baidu/hugegraph/loader/builder/EdgeBuilder.java
+++ b/src/main/java/com/baidu/hugegraph/loader/builder/EdgeBuilder.java
@@ -20,6 +20,7 @@
 package com.baidu.hugegraph.loader.builder;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.List;
@@ -46,6 +47,7 @@ public class EdgeBuilder extends ElementBuilder<Edge> {
     private final Collection<String> nonNullKeys;
     // Used to optimize access performace
     private VertexIdsIndex vertexIdsIndex;
+    private String[] lastNames;
 
     public EdgeBuilder(LoadContext context, InputStruct struct,
                        EdgeMapping mapping) {
@@ -69,9 +71,12 @@ public class EdgeBuilder extends ElementBuilder<Edge> {
 
     @Override
     public List<Edge> build(String[] names, Object[] values) {
-        if (this.vertexIdsIndex == null) {
+        if (this.vertexIdsIndex == null ||
+            (this.lastNames != names &&
+             !Arrays.equals(this.lastNames, names))) {
             this.vertexIdsIndex = this.extractVertexIdsIndex(names);
         }
+        this.lastNames = names;
         EdgeKVPairs kvPairs = this.newEdgeKVPairs();
         kvPairs.source.extractFromEdge(names, values,
                                        this.vertexIdsIndex.sourceIndexes);
@@ -142,7 +147,7 @@ public class EdgeBuilder extends ElementBuilder<Edge> {
                             "columns when id strategy is CUSTOMIZE");
         } else {
             throw new IllegalArgumentException(
-                      "Unsupported AUTOMATIC id strategy for hugegraph-loader");
+                    "Unsupported AUTOMATIC id strategy for hugegraph-loader");
         }
     }
 

--- a/src/main/java/com/baidu/hugegraph/loader/builder/EdgeBuilder.java
+++ b/src/main/java/com/baidu/hugegraph/loader/builder/EdgeBuilder.java
@@ -72,8 +72,7 @@ public class EdgeBuilder extends ElementBuilder<Edge> {
     @Override
     public List<Edge> build(String[] names, Object[] values) {
         if (this.vertexIdsIndex == null ||
-            (this.lastNames != names &&
-             !Arrays.equals(this.lastNames, names))) {
+            !Arrays.equals(this.lastNames, names)) {
             this.vertexIdsIndex = this.extractVertexIdsIndex(names);
         }
         this.lastNames = names;

--- a/src/main/java/com/baidu/hugegraph/loader/builder/EdgeBuilder.java
+++ b/src/main/java/com/baidu/hugegraph/loader/builder/EdgeBuilder.java
@@ -146,7 +146,7 @@ public class EdgeBuilder extends ElementBuilder<Edge> {
                             "columns when id strategy is CUSTOMIZE");
         } else {
             throw new IllegalArgumentException(
-                    "Unsupported AUTOMATIC id strategy for hugegraph-loader");
+                      "Unsupported AUTOMATIC id strategy for hugegraph-loader");
         }
     }
 

